gadget-container/entrypoint.sh

```diff
@@ -60,9 +60,11 @@ if grep -q '^1:name=systemd:.*/crio-[0-9a-f]*\.scope$' /proc/self/cgroup > /dev/
 fi
 
 # In the gadget image, /usr/src is a symlink to /host/usr/src (see gadget.Dockerfile).
-# If the kernel headers are already available on the gadget pod via the symlink,
-# no need to download them.
-if [ "$ID" = "rhcos" ] && [ ! -d "/usr/src/kernels/$KERNEL" ]; then
+# If the kernel headers are already available on the gadget pod via the
+# symlink, or the kheaders module is available no need to download them.
+if [ "$ID" = "rhcos" ] && [ ! -d "/usr/src/kernels/$KERNEL" ] && modprobe kheaders > /dev/null 2>&1; then
+  rrmod kheaders || true
```

Comment:
If the module was already loaded before the `modprobe` call, this would unload it...
Something like this keeps kheaders loaded/unloaded:
```
if grep -q kheaders /proc/modules || (sudo modprobe kheaders && sudo rmmod kheaders); then
  echo kheaders available
fi
```

Reply:
Updated to follow this logic, thanks for the suggestion!